python/ray/dataframe/dataframe.py

```diff
@@ -1321,7 +1323,8 @@ def _maybe_casted_values(index, labels=None):
                             values, mask, np.nan)
             return values
 
-        new_index = ray.get(_default_index.remote(new_obj)).index
+        _, new_index = _compute_length_and_index.remote(new_obj._df)
```

Comment:
The trouble here is that we're recomputing the lengths, but they should already be known. We can optimize this later.